### PR TITLE
workflows: add Python 3.11, Windows CLI tests to CI

### PR DESCRIPTION
**Pull request by zzzeid**
- add new build-and-test-windows-cli job (bug 1844111)
- skip test that fails on windows (bug 1844111)
- add Python 3.11 to test matrix (bug 1843853)

### CODE REVIEW

File: tests/unit/test_launchers.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from __future__ import absolute_import
 
 import os
+import sys
```
Comment (dklawren) [inline tests/unit/test_launchers.py:4]: Why was this added here?

**Review by cgsheeh — APPROVED** [review]

File: tests/unit/test_launchers.py (hunk 1 above)
Comment (zzzeid) [reply tests/unit/test_launchers.py:4]: `sys.platform` is used in lines 248 and 277.
